extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/KafkaIndexTaskClient.java

```diff
@@ -250,10 +251,50 @@ public DateTime getStartTime(final String id)
       return null;
     }
     catch (IOException e) {
-      throw Throwables.propagate(e);
+      throw new RuntimeException(e);
     }
   }
 
+  public Map<String, Object> getMovingAverages(final String id)
+  {
+    log.debug("GetMovingAverages task[%s]", id);
+
+    try {
+      final FullResponseHolder response = submitRequest(
+          id,
+          HttpMethod.GET,
+          "rowStats",
+          null,
+          true
+      );
+      return response.getContent() == null || response.getContent().isEmpty()
+             ? ImmutableMap.of()
+             : jsonMapper.readValue(response.getContent(), new TypeReference<Map<String, Object>>()
+             {
+             });
+    }
+    catch (NoTaskLocationException e) {
+      return ImmutableMap.of();
+    }
+    catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  public ListenableFuture<Map<String, Object>> getMovingAveragesAsync(final String id)
```

Comment:
Can this use java intrinsic futures? Does it need to be `ListenableFuture` or would a `Future` which is a `CompletableFuture` work here? setting the completable future in the `executorService` runnable would work

Reply:
hm, the other methods for supervisor -> task communication in there use ListenableFuture (which is also pretty common elsewhere in the codebase).

I'll leave this as ListenableFuture, I think using guava vs. java intrinsic futures is a larger decision that's out of scope for this PR.

Reply:
I think it's better to convert all ListenableFuture to CompletableFuture in another PR if needed.